crates/compiler/can/src/def.rs

```diff
@@ -2272,12 +2272,18 @@ fn canonicalize_pending_body<'a>(
 
     opt_loc_annotation: Option<Loc<crate::annotation::Annotation>>,
 ) -> DefOutput {
+    let mut loc_value = &loc_expr.value;
+
+    while let ast::Expr::ParensAround(value) = loc_value {
+        loc_value = value;
+    }
```

Comment:
@rtfeldman what do you think about unwrapping parens in a loop as the initial step? It's easy to refactor.
I'm not sure if it is worth refactoring the function to have a recursive approach. Am I missing something?

Reply:
Let me try another thing...

Reply:
Nah, it's not that simple and anyway requires separate refactoring.

From what I see, there's only tail call optimization for plain closure declarations, but not for ones that are inside of structures.

as a result, this fib works as expected
```
fib = \n, a, b ->
    if n == 0 then
        a
    else
        fib (n - 1) b (a + b)

fib 6 0 1
```

while this one fails because cannot find the recursion
```
fib = (
    \n, a, b ->
        if n == 0 then
            a
        else
            fib.0 (n - 1) b (a + b),
    1,
)

fib.0 6 0 1
```

Reply:
Created an issue: https://github.com/roc-lang/roc/issues/5636

Reply:
The loop approach seems fine to me! 👍